### DISCUSSION TITLE
How does integration with RustDesk work?

### DISCUSSION THREAD

**Discussion by jrbcast** [amidaware/tacticalrmm#1920] [General]
Hi, 

I am a licensed TeamViewer user providing IT support for my clients. Therefore, I have an extensive address book where I just click and connect to the client's computers. I was looking for a RMM solution when I found Tactical RMM and after trying it, I love it. Then I started to consider if I could change from TeamViewer to MeshCentral giving it is perfectly integrated into Tactical RMM. Nevertheless, using Mesh from the phone is somehow difficult at the moment (though it has improved as shown by release notes). Therefore, I have had a look at other alternatives and I see TRMM integrates with RustDesk but the free version of RustDesk does not include address book. 

My question is how does the integration work if using the free version? Is it the same as with Mesh that once you add a new agent it is also included and connecting is as easy as "Take Control" so it can act as an addressbook? Also, if I "click" on take control from an Android Phone (from the browser), will it open the native app to connect to the remote computer?

Thanks in advance, 
Jose

**Comment by dinger1986** [comment]
It uses a url action, I don't know on Android if it will work but it might.

Basically there's collector scripts for the details for RustDesk and that is saved to a custom field. It will work with pro or oss rustdesk

**Reply by jrbcast** [reply]
Thanks for your response. So, basically, it should work as it does now with Meshcentral? Just clicking on the take control?

**Reply by dinger1986** [reply]
No it would use a url action.

So right click url actions, RustDesk or could change the double click on the agent to run the url action to connect to rustdesk

**Comment by jrbcast** [comment]
Thanks a lot but I am afraid I am a bit lost with terminology :(. Its been
just 24 hours playing around with TRMM and I am not used to it. I have had
a look at docs but cannot find something explaining how everything
connects. For example,in this case, I understand I send a URL to the agent
but then....what happens on the agent side?bit opens a browser but how does
this translate into me connecting to that remote machine through Rustdesk?

Sorry for my lack of understanding:(.

**Reply by dinger1986** [reply]
https://docs.tacticalrmm.com/3rdparty_rustdesk/

Have you looked at that?

**Comment by jrbcast** [comment]
Yes, I have seen that. But I cannot understand how the whole thing works
(just to make sure it will cover my needs). I am currently with the first
script (I understand it is to be run on an agent of a machine I want to
control remotely) but I don't know, for example, what KEY= is or where to
find it...

Thanks a lot for your help

**Reply by dinger1986** [reply]
need to follow the guide how to install rustdesk first then you will have all that info.

**Comment by jrbcast** [comment]
Ok, you are right, I lost the key at the end of the script. I have managed
to create all scripts and tested that the installation one does work in a
test machine. Now I am lost with the other scripts and how to automate
them, specifically the step 5 of:

- Create collector tasks
<https://docs.tacticalrmm.com/functions/automated_tasks/#collector-tasks>
and Script Checks
<https://docs.tacticalrmm.com/functions/scripting/#script-checks>

I understand I must (or should to avoid doing it for each agent) create an
automation Policy. But from there I am lost on which script to invoke and
how... I suppose that after carefully following all docs and testing I will
get it!

Thanks!

**Comment by jrbcast** [comment]
I have set up everything and think I am getting done to it. The problem I
am facing now comes from the check task, it always return an error message
saying the test file does not exist. Nevertheless, I cannot see anywhere
where this file is created (it is not in the RuskDesk client installation
script either). When is this file created and how? Do I need to add
something to the script to enforce its creation or is the same installation
that will create it? I could add this to the installation script but don't
want to break anything:

$filePath = "C:\path\to\checkfile.txt" if (!(Test-Path $filePath)) {
New-Item -ItemType File -Path $filePath -Force }

Thanks, Jose

**Reply by dinger1986** [reply]
last line of this creates the file https://docs.tacticalrmm.com/3rdparty_rustdesk/#rustdesk-set-and-get-password-collector-script-needs-custom-agent-field-to-run-on-check-failure

**Comment by jrbcast** [comment]
And how or where do I need to set the trigger to launch that script in case
checking fails? I have created an automation and assigned it to my test
client but am totally lost on how to do it:

[image: imagen.png]

**Reply by dinger1986** [reply]
Create a task which runs on the check failure

https://docs.tacticalrmm.com/3rdparty_rustdesk/#rustdesk-set-and-get-password-collector-script-needs-custom-agent-field-to-run-on-check-failure
![trmm scheduled task](https://github.com/user-attachments/assets/8186af78-019b-49ca-bdc5-073db8ef0811)

**Comment by jrbcast** [comment]
Oh, nice, I am getting now a properly formed URL. Now it just doesn't work,
does not call anything... Will have a look at how to assign the URL for the
app. Thanks a lot!

**Comment by jrbcast** [comment]
Just an update. It did finally work. Just in case someone asks for Android.
It does open the Android APP which is great but asks for password, does not
take it from the URL. It is not a big deal, you can use the same password
for all agents being it hardcoded into the script (not advisable) or you
can just do a bit of hand work by entering the password for each new agent
and "remembering it".

Thanks a lot for your help, dinger1986!
